node/src/main/kotlin/net/corda/node/services/network/NodeInfoWatcher.kt

```
@@ -24,19 +23,20 @@ import kotlin.streams.toList
  * - Serialize and de-serialize a [NodeInfo] to disk and reading it back.
  * - Poll a directory for new serialized [NodeInfo]
  *
- * @param path the base path of a node.
- * @param pollFrequencyMsec how often to poll the filesystem in milliseconds. Any value smaller than 5 seconds will
+ * @param nodePath the base path of a node.
+ * @param pollFrequency how often to poll the filesystem in milliseconds. Any value smaller than 5 seconds will
  *        be treated as 5 seconds.
  * @param scheduler a [Scheduler] for the rx [Observable] returned by [nodeInfoUpdates], this is mainly useful for
  *        testing. It defaults to the io scheduler which is the appropriate value for production uses.
  */
+// TODO: Use NIO watch service instead?
 class NodeInfoWatcher(private val nodePath: Path,
-                      pollFrequencyMsec: Long = 5.seconds.toMillis(),
+                      pollFrequency: Duration = 5.seconds,
                       private val scheduler: Scheduler = Schedulers.io()) {
 
     private val nodeInfoDirectory = nodePath / CordformNode.NODE_INFO_DIRECTORY
-    private val pollFrequencyMsec: Long = maxOf(pollFrequencyMsec, 5.seconds.toMillis())
-    private val successfullyProcessedFiles = mutableSetOf<Path>()
+    private val pollFrequency: Duration = if (pollFrequency < 5.seconds) 5.seconds else pollFrequency
```

I didn't notice this. Better to throw an exception instead.